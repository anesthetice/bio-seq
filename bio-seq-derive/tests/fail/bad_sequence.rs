use bio_seq::prelude::*;

fn main() {
    let _seq = iupac!("ACGTx");
    let _seq = dna!("asdf");
}
