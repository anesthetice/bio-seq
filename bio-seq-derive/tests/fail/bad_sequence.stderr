error: Invalid IUPAC code at position 4: x
 --> tests/fail/bad_sequence.rs:4:16
  |
4 |     let _seq = iupac!("ACGTx");
  |                ^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `iupac` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Invalid DNA base at position 0: a
 --> tests/fail/bad_sequence.rs:5:16
  |
5 |     let _seq = dna!("asdf");
  |                ^^^^^^^^^^^^
  |
  = note: this error originates in the macro `dna` (in Nightly builds, run with -Z macro-backtrace for more info)
